brim/windows/launcher/zeek-launcher.go

```diff
@@ -0,0 +1,140 @@
+// +build windows
+
+// This tool executes zeek on windows, constructing the cygwin compatible ZEEK*
+// environment variables required.  It embeds knowledge of the locations of the
+// zeek executable and zeek script locations in the expanded 'zdeps/zeek'
+// directory inside a Brim installation.
+//
+// It uses the Windows "job object" api:
+// https://docs.microsoft.com/en-us/windows/win32/procthread/job-objects
+// to ensure that the launched zeek process is terminated when this process
+// exits.
+package main
+
+import (
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"unsafe"
+
+	"github.com/alexbrainman/ps"
+	"github.com/alexbrainman/ps/winapi"
+)
+
+// These paths are relative to the zdeps/zeek directory.
+var (
+	zeekExecRelPath  = "bin/zeek.exe"
+	zeekPathRelPaths = []string{
+		"share/zeek",
+		"share/zeek/policy",
+		"share/zeek/site",
+	}
+	zeekPluginRelPaths = []string{
+		"lib/zeek/plugins",
+	}
+)
+
+func cygPathEnvVar(name, topDir string, subdirs []string) string {
+	var s []string
+	for _, l := range subdirs {
+		p := filepath.Join(topDir, filepath.FromSlash(l))
+		vol := filepath.VolumeName(p)
+		cyg := "/cygdrive/" + vol[0:1] + filepath.ToSlash(p[len(vol):])
+		s = append(s, cyg)
+	}
+	val := strings.Join(s, ":")
+	return name + "=" + val
+}
+
+func launchZeek(zdepsZeekDir, zeekExecPath string, args []string) error {
+	zeekPath := cygPathEnvVar("ZEEKPATH", zdepsZeekDir, zeekPathRelPaths)
+	zeekPlugin := cygPathEnvVar("ZEEK_PLUGIN_PATH", zdepsZeekDir, zeekPluginRelPaths)
+
+	cmd := exec.Command(zeekExecPath, args...)
+	cmd.Stdin = os.Stdin
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	cmd.Env = append(os.Environ(), zeekPath, zeekPlugin)
+
+	return cmd.Run()
+}
+
+// ensureZeekTermination ensures that when this Go process terminates, the
+// launched Zeek process will also be terminated. It does so via the Windows
+// job objects api:
+// https://docs.microsoft.com/en-us/windows/win32/procthread/job-objects
+//
+// See this Go issue for discussion about the challenge of process management
+// on Windows, and the mention of the ps & winapi package used here:
+// https://github.com/golang/go/issues/17608
+func ensureZeekTermination() error {
```

Comment:
Nit: There's nothing specific to Zeek in here, so ensureSpawnedProcessTermination might be a better name.